monai/networks/nets/milmodel.py

```diff
@@ -203,42 +166,105 @@ def calc_head(self, x: torch.Tensor) -> torch.Tensor:
 
-            transformer_list = cast(nn.ModuleList, self.transformer)
+    def forward(self, x: torch.Tensor, no_head: bool = False) -> torch.Tensor:
 
-            x = transformer_list[0](l1)
-            x = transformer_list[1](torch.cat((x, l2), dim=2))
-            x = transformer_list[2](torch.cat((x, l3), dim=2))
-            x = transformer_list[3](torch.cat((x, l4), dim=2))
+        sh = x.shape
+        x = x.reshape(sh[0] * sh[1], sh[2], sh[3], sh[4])
+        if hasattr(self, "trans_pyramid_module"):
+            batch, channel = sh[0], sh[1]
+            x = self.trans_pyramid_module(x, batch=batch, channel=channel, no_head=no_head)
+        else:
+            x = self.net(x)
+            x = x.reshape(sh[0], sh[1], -1)
 
-            x = x.permute(1, 0, 2)
+            if not no_head:
+                x = self.calc_head(x)
 
-            a = self.attention(x)
-            a = torch.softmax(a, dim=1)
-            x = torch.sum(x * a, dim=1)
+        return x
 
-            x = self.myfc(x)
 
-        else:
-            raise ValueError("Wrong model mode" + str(self.mil_mode))
+class TransPyramidModule(nn.Module):
+    def __init__(
+        self,
+        num_classes: int,
+        backbone: nn.Module,
+        trans_blocks: int,
+        trans_dropout: float,
+        nfc: int,
+    ) -> None:
 
-        return x
+        super().__init__()
 
-    def forward(self, x: torch.Tensor, no_head: bool = False) -> torch.Tensor:
+        self.backbone = backbone
+        transformer_list = nn.ModuleList(
+            [
+                nn.TransformerEncoder(
+                    nn.TransformerEncoderLayer(d_model=256, nhead=8, dropout=trans_dropout), num_layers=trans_blocks
+                ),
+                nn.Sequential(
+                    nn.Linear(768, 256),
+                    nn.TransformerEncoder(
+                        nn.TransformerEncoderLayer(d_model=256, nhead=8, dropout=trans_dropout),
+                        num_layers=trans_blocks,
+                    ),
+                ),
+                nn.Sequential(
+                    nn.Linear(1280, 256),
+                    nn.TransformerEncoder(
+                        nn.TransformerEncoderLayer(d_model=256, nhead=8, dropout=trans_dropout),
+                        num_layers=trans_blocks,
+                    ),
+                ),
+                nn.TransformerEncoder(
+                    nn.TransformerEncoderLayer(d_model=2304, nhead=8, dropout=trans_dropout),
+                    num_layers=trans_blocks,
+                ),
+            ]
+        )
+        self.transformer = transformer_list
+        self.attention = nn.Sequential(nn.Linear(nfc, 2048), nn.Tanh(), nn.Linear(2048, 1))
+        self.myfc = nn.Linear(nfc, num_classes)
 
-        sh = x.shape
-        x = x.reshape(sh[0] * sh[1], sh[2], sh[3], sh[4])
+    def forward(self, x: torch.Tensor, batch: int, channel: int, no_head: bool = False):
+
+        x = self.backbone.conv1(x)  # type: ignore
```

Comment:
Hi @wyli , here I manually add `# type: ignore` for multiple lines, otherwise `mypy` will raise:
` error: "Tensor" not callable  [operator]`. Do you have any other suggestions to avoid doing like this? Thanks!

Reply:
have you tried to define the type when setting `self.backbone`? instead of here:https://github.com/Project-MONAI/MONAI/blob/94713c4ca7e1aa2a13db53b0c9fb19dc1f0a938b/monai/networks/nets/milmodel.py#L203

could try something like `self.backbone: model.ResNet = backbone  # type: ignore`...

essentially this type is not specific enough https://github.com/Project-MONAI/MONAI/blob/94713c4ca7e1aa2a13db53b0c9fb19dc1f0a938b/monai/networks/nets/milmodel.py#L195